src/lib/GifRequestDispatcher.ts

```diff
@@ -170,4 +171,78 @@ export class GifRequestDispatcher {
             error: "",
         };
     }
+
+    async syncGenerateGif(prompt: string): Promise<string | undefined> {
+        const apiUrl = await getSettingFromId(this.read, Preferences.API_URL);
+        const apiKey = await getSettingFromId(this.read, Preferences.API_KEY);
+        const modelId = await getSettingFromId(this.read, Preferences.MODEL_ID);
+
+        if (!apiKey || !apiUrl || !modelId) {
+            const errorMessage =
+                "ValidationError: One or more preferences are not set";
+            this.app.getLogger().log(errorMessage);
+            return undefined;
+        }
+
+        const requestBody: IGifRequestBody = {
+            version: modelId,
+            input: {
+                mp4: false,
+                steps: 30,
+                width: 256,
+                height: 256,
+                prompt,
+                negative_prompt: "blurry",
+            },
+        };
+
+        const headers = {
+            Authorization: `Bearer ${apiKey}`,
+            "Content-Type": "application/json",
+        };
+
+        const res = await this.http.post(apiUrl, {
+            headers: headers,
+            data: requestBody,
+        });
+
+        const genResponse = res.data as IGifResponseData;
+
+        let status: PredictionStatus = PredictionStatus.STARTING;
+        let output: string | undefined;
+
+        while (
+            status !== PredictionStatus.SUCCEEDED &&
+            status !== PredictionStatus.FAILED &&
+            status !== PredictionStatus.CANCELLED
```

Comment:
Instead of have this you can use includes, Following: 

```suggestion
 ```
    const invalidStatuses = [
    PredictionStatus.SUCCEEDED,
    PredictionStatus.FAILED,
    PredictionStatus.CANCELLED]; 
     while(!invalidStatuses.includes(status)) {
       // Your code here
     }